### PR TITLE
Sonic '06 PKG Reading/Writing

### PR DESCRIPTION
**Pull request by HyperPolygon64**
Added reading and writing support for the PKG format.

### CODE REVIEW

File: HedgeLib/Misc/S06Package.cs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            writer.Write((uint)Files.Count);
+            writer.AddOffset("fileEntriesPos");
+            writer.Write((uint)Types.Count);
+            writer.AddOffset("typeEntriesPos");
+
+            // Types
+            for (int i = 0; i < Types.Count; i++)
+            {
+                if (i == 0)
+                    writer.FillInOffset("typeEntriesPos", false);
```
Comment (Radfordhound) [inline HedgeLib/Misc/S06Package.cs:111]: No need to check if (i == 0) repeatedly for every item in the loop - just fill-in the offset once before the loop:

```suggestion
            writer.FillInOffset("typeEntriesPos", false);
            for (int i = 0; i < Types.Count; i++)
            {
```

File: HedgeLib/Misc/S06Package.cs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                writer.AddOffset($"typeFilesOffset{i}");
+            }
+
+            // Files
+            uint fileEntriesPos = (uint)writer.BaseStream.Position;
+
+            for (int i = 0; i < Files.Count; i++)
+            {
+                if (i == 0)
+                    writer.FillInOffset("fileEntriesPos", false);
```
Comment (Radfordhound) [inline HedgeLib/Misc/S06Package.cs:124]: Same thing here

Comment (HyperPolygon64) [reply HedgeLib/Misc/S06Package.cs]: Good point, resolved both to be outside the loop. 👍

**Comment by Radfordhound** [comment]
Thanks! Yeah that's happened because of a tab character that somehow got introduced in a file that has spaces elsewhere. Not a big deal though.

**Pull request merged**